config/listenaddress/listenaddress.go

```diff
@@ -42,59 +45,60 @@ const (
 
 // Configuration is the configuration for resolving a listen address.
 type Configuration struct {
-	// Hostname is the hostname to use
-	Hostname string `yaml:"hostname" validate:"nonzero"`
-
-	// Port specifies the port to listen on
-	Port *Port `yaml:"port" validate:"nonzero"`
-}
-
-// Port specifies the port to use
-type Port struct {
 	// PortType is the port type for the port
 	PortType Resolver `yaml:"portType" validate:"nonzero"`
```

Comment:
Maybe we should rename this to just yaml:”type” to match the host ID resolver yaml type field?

Reply:
Also since this is actually a different way of specifying a listen address (possibly port only) maybe we should rename the type from PortType to ListenAddressType?